Core/HLE/sceFont.cpp

```diff
@@ -427,11 +498,13 @@ class FontLib {
 	}
 
 	void Done() {
+		bool needPending = false;
 		for (size_t i = 0; i < fonts_.size(); i++) {
 			if (isfontopen_[i] == FONT_IS_OPEN) {
-				fontMap[fonts_[i]]->Close();
+				CloseFont(fontMap[fonts_[i]]);
 				delete fontMap[fonts_[i]];
 				fontMap.erase(fonts_[i]);
+				needPending = true; // Pending all the next mipcalls.
```

Comment:
Because the callback may be called mutipletimes in a syscall, and the former will be covered by the latter, so put the latter calls to the pending list.

Reply:
Okay, what if we just change this:

```c++
static bool __CanExecuteCallbackNow(Thread *thread) {
	return g_inCbCount == 0;
}
```

To:

```c++
static bool __CanExecuteCallbackNow(Thread *thread) {
	return currentCallbackThreadID == 0 && g_inCbCount == 0;
}
```

Does that also make it work?  I think I'd rather avoid the syscall having to decide whether it should be pending or not.

-[Unknown]